### ISSUE TITLE
Issue installing and running ivis R package in RStudio

### ISSUE THREAD

**Issue by kevinrue** [beringresearch/ivis#28]
Hello,

For [JOSS review](https://github.com/openjournals/joss-reviews/issues/1596).

The installation instructions fail when run in the RStudio environment:

```
> devtools::install_github("beringresearch/ivis/R-package", force=TRUE)
Downloading GitHub repo beringresearch/ivis@master
✔  checking for file ‘/private/var/folders/cp/8rn2cs_x79zcbp_yb75ychg80000gq/T/Rtmpud6pnU/remotesbe4d59017fdb/beringresearch-ivis-bbccdb7/R-package/DESCRIPTION’ ...
─  preparing ‘ivis’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘ivis_1.1.3.tar.gz’
   
* installing *source* package ‘ivis’ ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘ivis’:
 .onLoad failed in loadNamespace() for 'ivis', details:
  call: path.expand(path)
  error: invalid 'path' argument
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Library/Frameworks/R.framework/Versions/3.6/Resources/library/ivis’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/3.6/Resources/library/ivis’
Error: Failed to install 'ivis' from GitHub:
  (converted from warning) installation of package ‘/var/folders/cp/8rn2cs_x79zcbp_yb75ychg80000gq/T//Rtmpud6pnU/filebe4d71713083/ivis_1.1.3.tar.gz’ had non-zero exit status
```

However, it does work fine when run in the console (Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64):

```
> devtools::install_github("beringresearch/ivis/R-package", force=TRUE)
Downloading GitHub repo beringresearch/ivis@master
   checking for file ‘/private/var/folders/cp/8rn2cs_x79zcbp_yb75ychg80000gq/T/Rtmpvj2CT3/remotesc3827327cfb8/beringresearch-ivis-bbccdb7/R-package/DESCRIPTION’✔  checking for file ‘/private/var/folders/cp/8rn2cs_x79zcbp_yb75ychg80000gq/T/Rtmpvj2CT3/remotesc3827327cfb8/beringresearch-ivis-bbccdb7/R-package/DESCRIPTION’
─  preparing ‘ivis’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘ivis_1.1.3.tar.gz’
   
* installing *source* package ‘ivis’ ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (ivis)
```

Moreover, the `ivis` package (installed from the terminal) can be loaded from an R console in a terminal, but throws the following error when loaded in RStudio

```
> library(ivis)
Error: package or namespace load failed for ‘ivis’:
 .onLoad failed in loadNamespace() for 'ivis', details:
  call: path.expand(path)
  error: invalid 'path' argument
```

This is most likely due to `conda` not being on the `PATH` in RStudio:
```
# RStudio
> system("echo $PATH")
/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/ncbi/igblast/bin:/Library/TeX/texbin:/opt/X11/bin:/opt/local/bin
# Console
> system("echo $PATH")
/Users/kevin/miniconda3/bin:/Users/kevin/miniconda3/condabin:/usr/local/opt/imagemagick@6/bin:/Users/kevin/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/ncbi/igblast/bin:/Library/TeX/texbin:/opt/X11/bin
```

Is there a recommended way to set up an environment to run `ivis` in RStudio, or are users only expected to run it from a terminal R console?

Thanks!

**Comment by idroz** [comment]
Hi - thanks for trying out `ivis` and testing it out in RStudio. It does sound like a path issue. I haven't worked much with RStudio recently, but it may be that RStudio doesn't inherit system $PATH: https://stackoverflow.com/questions/31121645/rstudio-shows-a-different-path-variable

I'm thinking that a potential solution would be to get rid of Conda manager altogether and simply install R wrapper through virtualenv + pip. I'll test this out over the next few days and revert here.

Thanks again!

**Comment by kevinrue** [comment]
To be fair, I am not an expert with regard to `reticulate` yet, so there may be some trick to fix the situation that I don't know. I'd recommend looking at other packages that use `reticulate` to see how they deal with this kind of situation, if you can find some.
I look forward to testing the method directly within an R environment, but in the meantime I'm happy to test the functionality in a Python environment.
Good luck!

**Comment by idroz** [comment]
Hi again - I made a few changes to `ivis` install scripts. They now use `virtualenv` rather than Conda. I've also caught up R wrapper with the latest stable version of `ivis` on PyPi. Both installation and demo scripts seem to work inside RStudio and on command line. Could you please test out the development branch:

```
devtools::install_github("beringresearch/ivis/R-package", ref="r-virtualenv-install")
```

Subsequent are the same as in master branch README.
Thanks!

**Comment by kevinrue** [comment]
After updating my installation using your `devtools`  command above, the following worked for me now, well done!
```
library(ivis)

model <- ivis(k = 3, batch_size = 3)

X = data.matrix(iris[, 1:4])
model = model$fit(X)

xy = model$transform(X)
```

**Comment by kevinrue** [comment]
I'll reopen and let you close the issue when you merge the update in a PR.

**Comment by idroz** [comment]
Addressed in #30
